core/frontend/src/IModelConnection.ts

```diff
@@ -1101,15 +1141,11 @@ export namespace IModelConnection {
      * @throws [[IModelError]] if the Id is invalid or if no CodeSpec with that Id could be found.
      */
     public async getById(codeSpecId: Id64String): Promise<CodeSpec> {
+      if (codeSpecId === "") throw new IModelError(IModelStatus.NotFound, "CodeSpec not found");
       if (!Id64.isValid(codeSpecId))
         throw new IModelError(IModelStatus.InvalidId, "Invalid codeSpecId", () => ({ codeSpecId }));
 
-      await this._loadAllCodeSpecs(); // ensure all codeSpecs have been downloaded
-      const found: CodeSpec | undefined = this._loaded!.find((codeSpec: CodeSpec) => codeSpec.id === codeSpecId);
-      if (!found)
-        throw new IModelError(IModelStatus.NotFound, "CodeSpec not found");
-
-      return found;
+      return this._loadCodeSpec({ id: codeSpecId, name: undefined as never });
```

Comment:
You can avoid being forced to assign value to `name` by making `name` optional in `_loadCodeSpec` definition (`name: never` -> `name?: never`). Same with `id`

Reply:
Only `undefined` should work. 

Removing it completely would allow passing both `id` and `name` at the same time which is not ideal even for private method.